datalad_registry/tasks.py

```diff
@@ -112,26 +174,13 @@ def collect_dataset_info(
     # I guess they might need to be groupped by ds_id since the same
     # cache location is to be reused - each task should then handle it
     for (ds_id, url) in datasets:
-        ds_path = cache_dir / ds_id[:3] / url_encode(url)
-        ds_path_str = str(ds_path)
-        # TODO (later): Decide how to handle subdatasets.
-        # TODO: support multiple URLs/remotes per dataset
-        # Make remote name to be a hash of url. Check below if among
-        # remotes and add one if missing, then use that remote, not 'origin'
-        if ds_path.exists():
-            ds = dl.Dataset(ds_path)
-            ds_repo = ds.repo
-            ds_repo.fetch(all_=True)
-            ds_repo.call_git(["reset", "--hard", "refs/remotes/origin/HEAD"])
-        else:
-            ds = dl.clone(url, ds_path_str)
-            ds_repo = ds.repo
-        info: InfoType = _extract_git_info(ds_repo)
-        info.update(_extract_annex_info(ds_repo))
+        abbrev_id = "None" if ds_id is None else ds_id[:3]
+        ds_path = cache_dir / abbrev_id / url_encode(url)
+        ds = clone_dataset(url, ds_path)
+        info = get_info(ds.repo)
+        if ds_id is None:
+            info["ds_id"] = ds.id
```

Comment:
Unlikely but ds_id can change... I think we should eventually handle those cases (rename directory etc), but for now I guess it could suffice to alert that we ran into such one
```suggestion
        if ds_id is None:
            info["ds_id"] = ds.id
        elif ds_id != ds.id:
            lgr.warning("A dataset with an ID (%s) got a new one (%s)", ds_id, ds.id)
```